src/library_wasm_worker.js

```diff
@@ -98,8 +99,8 @@ addToLibrary({
     // Run the C side Worker initialization for stack and TLS.
     _emscripten_wasm_worker_initialize(m['sb'], m['sz']);
 #if PTHREADS
-    // Record that this Wasm Worker supports synchronous blocking in emscripten_futex_wake().
-    ___set_thread_state(/*thread_ptr=*/0, /*is_main_thread=*/0, /*is_runtime_thread=*/0, /*supports_wait=*/0);
+    // Record that this Wasm Worker supports synchronous blocking in emscripten_futex_wait().
```

Comment:
Maybe just remove this comment line.   The name `workerSupportsFutexWait` is self explanitiory and the comment is not accurate since this Wasm Worker doesn't always support blocking.